pykechain/models/part.py

```diff
@@ -717,3 +718,154 @@ def populate_descendants(self, batch=200):
                     parent._cached_children.append(child)
 
         self._cached_children = descendants_flat_list[0]._cached_children
+
+    def clone(self, **kwargs):
+        parent = self.parent()
+        return self._client._create_clone(parent, self, **kwargs)
+
+    def copy_model(self, target_parent, name=None, include_children=True, include_instances=True):
+        """
+        Copy the `Part` model to target parent, which also needs to be a model.
+
+        .. versionadded:: 2.3
+
+        :param target_parent: `Part` object under which the desired `Part` is copied
+        :type target_parent: :class:`Part`
+        :param name: how the copied top-level `Part` should be called
+        :type name: basestring
+        :param include_children: True to copy also the descendants of `Part`.
+        :type include_children: bool
+        :param include_instances: True to copy also the instances of `Part` to ALL the instances of target_parent.
+        :type include_instances: bool
+        :returns: copied :class:`Part` model.
+        :raises IllegalArgumentError: if part and target_parent have different `Category` than MODEL
+        :raises IllegalArgumentError: if part and target_parent are identical
+
+        Example
+        -------
+        >>> model_to_copy = client.model(name='Model to be copied')
+        >>> bike = client.model('Bike')
+        >>> model_to_copy.copy_model(target_parent=bike, name='Copied model', include_children=True,
+        >>>                          include_instances=True)
+
+        """
+        if self.id == target_parent.id:
+            raise IllegalArgumentError('Cannot copy model "{}" with id "{}" under itself.'.format(self.name, self.id))
+        copied_model = relocate_model(part=self, target_parent=target_parent, name=name,
+                                      include_children=include_children)
+        if include_instances:
+            instances_to_be_copied = list(self.instances())
+            parent_instances = list(target_parent.instances())
+            for parent_instance in parent_instances:
+                for instance in instances_to_be_copied:
+                    instance.populate_descendants()
+                    move_part_instance(part_instance=instance, target_parent=parent_instance,
+                                       part_model=self, name=instance.name, include_children=include_children)
+
+        return copied_model
+
+    def move_model(self, target_parent, name=None, include_children=True, include_instances=True):
```

Comment:
can't we just use `move()` and if it is an instance or model do the math in the helpers?